automation/orchestrator.go

```diff
@@ -36,44 +36,43 @@ type OrchestratorOptions struct {
 	RunStore      run.Store
 }
 
-// Run persists the dataset that results from executing a workflow transform
-type Run func(ctx context.Context, streams ioes.IOStreams, w *workflow.Workflow, runID string) error
-
-// RunFactory is a function that produces a Run function
-type RunFactory func(ctx context.Context) Run
-
-// Apply executes an ephemeral workflow transform
-type Apply func(ctx context.Context, wait bool, runID string, w *workflow.Workflow, ds *dataset.Dataset, secrets map[string]string) error
+// WorkflowRunner is for running workflows using some execution engine
+// only non-test implementation is lib.Instance, but this interface is used
+// to avoid a direct dependency
+type WorkflowRunner interface {
+	RunEphemeral(context.Context, string, *workflow.Workflow, *dataset.Dataset, bool, WorkflowRunParams) error
+	RunAndCommit(context.Context, string, *workflow.Workflow, ioes.IOStreams, WorkflowRunParams) error
+}
 
-// ApplyFactory is function that produces an Apply function
-type ApplyFactory func(ctx context.Context) Apply
+// WorkflowRunParams are additional parameters for a workflow run
+type WorkflowRunParams struct {
+	Secrets      map[string]string
+	OutputWidth  int
+	OutputHeight int
```

Comment:
I'm into the declaration of the `WorkflowRunParams` struct. Another possible place to put these fields is on `ioes.IOStreams`